### ISSUE TITLE
rgbToHsb() fails on input containing { red: 255 }

### ISSUE THREAD

**Issue by emcmanus** [Shopify/polaris#251]
# Issue summary

`rgbToHsb()` fails on any input containing `{ red: 255 }`

I think the error is related to how `rgbToHsb` calculates hue when `red` is the largest color value.

## Expected behavior

```js
rgbToHsb({red: 255, green: 0, blue: 183, alpha: 1})
```

This should return `{ hue: 317, ... }`, which you can calculate using http://www.workwithcolor.com/color-converter-01.htm

## Actual behavior

```js
rgbToHsb({red: 255, green: 0, blue: 183, alpha: 1})
```

This incorrectly returns `{ hue: 0, ... }`

## Steps to reproduce the problem

1. `import { rgbToHsb } from "@shopify/polaris";`
1. Call `rgbToHsb()` with a value containing `{ red: 255 }`

## Specifications

- Polaris version: 1.9.1 (present since v1.0.1)
- React version: N/A
- Browser: N/A
- Device: N/A
- Operating System: N/A

---

⚓️ We’re not accepting pull requests at this time<br />
🗒 This repo is for reporting issues and feature requests only

**Comment by emcmanus** [comment]
Here's the fix https://github.com/Shopify/polaris/commit/e2a72fe639b424a30160e9960782e7d5acf5c887

**Comment by emcmanus** [comment]
Incidentally by rounding `hue` before the return you introduce precision loss that prevents a proper mapping between HSB and RGB.

**Comment by dfmcphee** [comment]
Hey @emcmanus, thanks for the issue. This should now be fixed as of 1.10.0.
